src/server.c

```diff
@@ -3888,7 +3888,13 @@ static void readOOMScoreAdj(void) {
  * A process_class value of -1 implies OOM_CONFIG_MASTER or OOM_CONFIG_REPLICA,
  * depending on current role.
  */
+static int oom_adj_managed_by_redis = 0;
 int setOOMScoreAdj(int process_class) {
+
+    if (server.oom_score_adj != OOM_SCORE_ADJ_NO && !oom_adj_managed_by_redis) {
+        readOOMScoreAdj();
```

Comment:
maybe this is the opportunity to eliminate this method and just embed it here?
it modifies a global value (oom_score_adj_base) and doesn't return anything, so it's name is odd, and also doesn't have any doc comment.
since the oom_score_adj_base var is only used here, maybe that makes sense to move everything here.
or even make that var a static var of the function too, rather than a global.

the only concerns i have is if there was some specific reason for which @yossigo decide to read it at startup.
and if how we handle read failure (currently it just returns without modifying the var.

Reply:
This makes sense but I now realize I did a change here. Now I read the base value each time we enable `oom-score-adj` and not on startup. This means that if the user starts Redis and then changes the oom score manually and then enables `oom-score-adj` then I'll read the **current** oom score as the base and not the value on startup. This makes more sense to me but I'd like to hear your/@yossigo's feedback.
Another example of the change is if the user wants to modify the base, they can now disable `oom-score-adj`, then manually update _/proc_ and then re-enable. This will create a new base.
I also think since the base is used as a sort of configuration modifier it'll be good to leave it in the `server` struct, maybe in the future we'll add it to `INFO` or something.

Reply:
i thought it is intentional, and it does make sense to me.
also, i don't think it's a likely situation (both us and the user mess with `/proc`).
my only concern is that maybe there's something inherently different between reading it on startup vs later (some system issue).

regarding the variable, i do think that if we keep updating it every time we're enabled, and the only method that touches it is this one, i do think it should not be in the server struct (as an evidence, here in your PR you're changing it's meaning)